### DISCUSSION TITLE
Request body and json documentation

### DISCUSSION THREAD

**Discussion by bitdom8** [oven-sh/bun#2715] [General]
Documentation doesn't cover the text and json part with `async`. So here it is:

 ```
 async fetch(req) {
    const url = new URL(req.url);
    if (url.pathname === "/") return new Response(`Home page!`);
    if (url.pathname === "/w/wht") {
      JSON.stringify(await req.json())
     SON.stringify(await req.text())
      return new Response("Blog!"
        , {
          headers: {
            "content-type": "*/*",
            type: '*/*'
          }
        }
      )
    };
}
```

**Comment by bitdom8** [comment]
Seems `async fetch(req)` causes troubles in terms of server running with Telegram bots

**Reply by tomByrer** [reply]
> Seems `async fetch(req)` causes troubles in terms of server running with Telegram bots

How so?

**Comment by bitdom8** [comment]
I include try catch in each of the URL 's. That was the cause. Now I make only one try and catch all is well:

`Bun.serve({
  port: 8765,
  async fetch(req) {
    try {
if (req.method !== "POST") {
        return new Response(null, { status: 405 });
      }`
      
     
Thanks for saving us from node @tomByrer
